### PR TITLE
[WIP]  HIVE-24390 Spelling

### PR DESCRIPTION
**Pull request by jsoref**
Note: GitHub objects to PRs that exceed 250 commits, this PR has 734. I don't expect it to be accepted as is. I'm currently trying to access my Apache account.

Generally, projects choose to split PRs like this into pieces, sometimes by module, sometimes by change type.

### What changes were proposed in this pull request?

This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

### Why are the changes needed?

Typos lead to all sorts of problems, branding errors, tests that don't test what's expected, difficulty in understanding both the code and the product

### Does this PR introduce _any_ user-facing change?

Yes. Considering the scope of this PR, I'll update this later.

### How was this patch tested?

The misspellings have been reported at https://github.com/jsoref/hive/commit/9f53c869b2494f9d865a8db77c569277e3509d54#commitcomment-44193420

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/hive/commit/0c87b35337562d845312ef3c3a06ff8eb2295c57

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

**Comment by jsoref** [comment]
Sigh, I'll try to rebase this weekend

**Comment by jsoref** [comment]
Sigh

**Comment by jsoref** [comment]
I'm not sure what to do about this.

File: ql/src/test/queries/clientnegative/dyn_part_empty.q.disabled [hunk 1]
```diff
@@ -15,7 +15,7 @@
 -- limitations under the License.
 
 set hive.exec.dynamic.partition=true;
-set hive.stats.autogether=false;
+set hive.stats.autogather=false;
```
Comment (kgyrtkirk) [inline ql/src/test/queries/clientnegative/dyn_part_empty.q.disabled:18]: lol!

File: service/lib/php/ext/thrift_protocol/php_thrift_protocol.cpp [hunk 2]
```diff
@@ -429,7 +429,7 @@ void binary_deserialize(int8_t thrift_typeID, PHPInputTransport& transport, zval
         throw_tprotocolexception(errbuf, INVALID_DATA);
         RETURN_NULL();
       }
-      binary_deserialize_spec(return_value, transport, Z_ARRVAL_P(spec));
+      binary_deserialize_spec(return_value, transport, Z_ARRIVAL_P(spec));
```
Comment (kgyrtkirk) [inline service/lib/php/ext/thrift_protocol/php_thrift_protocol.cpp:432]: have you regenerated the thrift interfaces? I'm not sure if this is coming from Hive

**Review by kgyrtkirk — COMMENTED** [review]
looks good - would it be possible to enable the checker to run it on every PR?

File: service/lib/php/ext/thrift_protocol/php_thrift_protocol.cpp (hunk 2 above)
Comment (jsoref) [reply service/lib/php/ext/thrift_protocol/php_thrift_protocol.cpp:432]: oops, thanks, this is just a botch ARRVAL is presumably short for array-val. dropping it and rebasing ...

**Comment by jsoref** [comment]
It's definitely possible to run on every PR. Let me see about updating this PR. (We'd want to merge everything through Z, and do a new run for drift that occurred before enabling it.)

File: ql/src/test/queries/clientnegative/dyn_part_empty.q.disabled (hunk 1 above)
Comment (jsoref) [reply ql/src/test/queries/clientnegative/dyn_part_empty.q.disabled:18]: Oh, you weren't telling me to drop this. Sigh. (Restored)

Yes, typos can be funny. But figuring out if they're intended is hard. (Earlier when I was looking at this comment, I parsed it at laughing at an incorrect correction!)

Comment (kgyrtkirk) [reply ql/src/test/queries/clientnegative/dyn_part_empty.q.disabled:18]: not at all :) `hive.stats.autgether` is a non-existent feature toggle :)
but this test seems to be disabled - so its kinda in the gray zone

**Comment by kgyrtkirk** [comment]
you could add the cheker in a separate PRs as well - this ticket already has a lot of stuff

**Comment by jsoref** [comment]
Yeah, I definitely wouldn't do it in the initial rounds. Once I have what appears to be a clean base, I'll make a PR for it -- typically a handful of extra misspellings sneak in between when I do the initial work and when I make the PR for the checker, and those should be pulled in at the same time, but.. for this repo, that'll be a while off anyway.

On the bright side, it looks like it's making progress, thanks for the pointer.

**Review by kgyrtkirk — APPROVED** [review]
this patch already has a lot of changes - I think it would make sense to get it in and  continue in a new PR? what do you think?

**Comment by jsoref** [comment]
Yes. That's my plan. This is sadly a tiny slice. I think I'll do another PR of about the same size. When we get to the end, I'll see about a PR for the action-workflow.

**Comment by kgyrtkirk** [comment]
@jsoref sounds good ; I've merged the PR before it gets into a conflict - we could always open more.

I think probably the best would be to start in one of the corners of the project - like in the `common` or `serde` module; and fix and enable the automatic checker on that portion...that way new commits will not have a chance to erode the parts which were already covered...what do you think about this approach?

**Comment by jsoref** [comment]
That's definitely possible. Sure. I'm not sure I'll have time before next week, but, I can do a slice of one or two of them.

**Pull request merged**